src/AI/Console/AddDocBlocksCommand.php

```diff
@@ -2,45 +2,64 @@
 
 declare(strict_types=1);
 
-namespace BumbleDocGen\Console\Command;
+namespace BumbleDocGen\AI\Console;
 
+use BumbleDocGen\AI\Traits\SharedCommandLogicTrait;
+use BumbleDocGen\Console\Command\BaseCommand;
 use BumbleDocGen\Core\Configuration\Exception\InvalidConfigurationParameterException;
 use BumbleDocGen\LanguageHandler\Php\Parser\Entity\Exception\ReflectionException;
 use DI\DependencyException;
 use DI\NotFoundException;
+use GuzzleHttp\Exception\GuzzleException;
+use JsonException;
 use Symfony\Component\Console\Input\InputInterface;
 use Symfony\Component\Console\Output\OutputInterface;
-use Tectalic\OpenAi\ClientException;
 
-final class FillInReadmeMdTemplateCommand extends BaseCommand
+final class AddDocBlocksCommand extends BaseCommand
 {
+    use SharedCommandLogicTrait;
+
+    public const NAME = 'ai:add-doc-blocks';
+
     protected function getCustomConfigOptionsMap(): array
     {
         return [
             'project_root' => 'Path to the directory of the documented project',
             'templates_dir' => 'Path to directory with documentation templates',
             'cache_dir' => 'Configuration parameter: Path to the directory where the documentation generator cache will be saved',
+            'ai_provider' => 'The AI service to use, options: openai',
+            'ai_api_key' => 'The API key to use when interacting with the AI',
```

Comment:
What if the key is not required in the provider but other parameters are required? I think that then it’s worth getting the remaining parameters except the provider in the provider factory method